packages/backend/src/QueryEngine.ts

```diff
@@ -96,8 +133,13 @@ export class QueryEngine<DB> {
         ];
     }
 
-    execute<TTable extends Table<DB>, TQuery extends Query<DB, TTable>>(
+    execute<
+        TTable extends Table<DB>,
+        TQuery extends Query<DB, TTable>,
+        TContext,
+    >(
         query: TQuery,
+        context?: TContext,
```

Comment:
context should be part of `opts`

Reply:
Generally speaking I try not to have too many function arguments. Why so? readability.

```ts
updateUser(123,false)
```

is way less readable than

```ts
updateUser({id:123, is_active:false})
```

If you ever get to work with a codebase that abuses positional parameters you will know what a hell it is to call 

```ts
sendEvent("user.registered",1,2,new Date(),true,true,false)
```

And how easy it is to accidentally mistype something.

Reply:
It's also easier to refactor: you can see it on this very MR: just notice all the work you had to do with refactoring the `queryEngine.execute` calls

```
return queryEngine.execute(query, {}, { returnLastOnly });
                                                            ^
                                                             |    
```

It's not necessary, just do

```ts
return queryEngine.execute(query, { returnLastOnly, context?: Context });
```